pywbem/_mof_compiler.py

```diff
@@ -2672,7 +2674,15 @@ def compile_string(self, mof, ns, filename=None):
         except AttributeError:
             oldmof = None
         self.parser.mof = mof
+
+        # Save default_namepace so it can be restored after the parse.
+        # Note that there may be nested calls to compile_string
+        try:
+            oldns = self.parser.handle.default_namespace
+        except AttributeError:
+            oldns = None
```

Comment:
This approach of handling the AttributeError in the other cases of `self.parser.file` and `self.parser.mof` is just a very slacky way of dealing with not properly having thought about initialization. The `self.parser.handle` is in fact properly initialized, so that is not necessary there. In fact, it creates a case of non-testability for the exception path.

I suggest to remove the exception handling for this case.

Beyond that, we could also rework the other two cases to use proper initialization rather than exception handling.

Reply:
DISCUSSION:  I must be missing something because I am not certain what the exception gains but we have to be careful in this method is recursively called multiple times, not directly but as a result of the calls to the parser which may end up starting another compile. In my tests we were at points 4 levels into recursive calls to compile_string.

Reply:
I dropped the additions.  I want to discuss the original use since I am not sure why needed at all **DISCUSSION**  Those variables should be private to every nested call as I understand nested method calls in python.

Reply:
I'm ok not removing the exception handling in the other two cases (`self.parser.file` and `self.parser.mof`) for now, since they are unrelated to this PR. So we are DONE with this comment.